### PR TITLE
Proj0800 & Proj081: CPM

### PR DESCRIPTION
**Pull request by Corniel**
Documentation for: https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers/pull/145

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,12 +79,22 @@ reported a the [GibHub repository](https://github.com/dotnet-project-file-analyz
...
 * [**Proj0241** Enable package baseline validation](rules/Proj0241.md)
 * [**Proj0242** Generate NuGet packages conditionally](rules/Proj0242.md)
-* [**Proj0400** Define the project publishability explicitly](rules/Proj0400.md)
 * [**Proj0600** Avoid generating packages on build if not packable](rules/Proj0600.md)
+
+### Publishing
+* [**Proj0400** Define the project publishability explicitly](rules/Proj0400.md)
+
+### Central Package Version Management
+* [**Proj0800** Configure Central Package Version Management](rules/Proj0800.md)
```
Comment (LauraXiulan) [inline README.md:88]: Proj0800 twice?

File: rules/Proj0800.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Proj0800: Configure Central Package Version Management explicitly
+Dependency management is a core feature of NuGet. Managing dependencies for a
+single project can be easy. Managing dependencies for multi-project solutions
+can prove to be difficult as they start to scale in size and complexity. In
+situations where you manage common dependencies for many different projects,
+you can leverage NuGet's central package management (CPM) features to do all of
+this from the ease of a single location.
+
+Hence, this rules advises to enable Central Package Version Management via
```
Comment (LauraXiulan) [inline rules/Proj0800.md:9]: "by" instead of "via"

File: rules/Proj0800.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+enabling it with `<DevelopmentDependency>true</DevelopmentDependency>`.
+
+## Non-complaint
```
Comment (LauraXiulan) [inline rules/Proj0800.md:12]: compl**ia**nt

File: rules/Proj0800.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# Proj0800: Configure Central Package Version Management explicitly
+Dependency management is a core feature of NuGet. Managing dependencies for a
+single project can be easy. Managing dependencies for multi-project solutions
+can prove to be difficult as they start to scale in size and complexity. In
+situations where you manage common dependencies for many different projects,
+you can leverage NuGet's central package management (CPM) features to do all of
```
Comment (LauraXiulan) [inline rules/Proj0800.md:6]: Is it CPM or CPVM? You are using both

File: rules/Proj0800.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+this from the ease of a single location.
+
+Hence, this rules advises to enable Central Package Version Management via
+enabling it with `<DevelopmentDependency>true</DevelopmentDependency>`.
```
Comment (Sjaaky) [inline rules/Proj0800.md:10]: After reading https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management, https://github.com/NuGet/Home/wiki/DevelopmentDependency-support-for-PackageReference and https://github.com/NuGet/Home/issues/4125 I don't how the 2 concepts relate to each other.
I do see a suggestion for Proj1101 Package references should have stable versions, which shouldn't be triggered for DevelopmentDependencies.

File: rules/Proj0801.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Proj0801: Include 'Directory.Packages.props'
+When [Central Package Version Management](Proj0800.md) the
```
Comment (LauraXiulan) [inline rules/Proj0801.md:2]: ... is enabled the ...

**Review by Sjaaky — CHANGES_REQUESTED** [review]
I see 2 concepts mixed, or am I mistaken?

File: rules/Proj0800.md (hunk 5 above)
Comment (LauraXiulan) [inline rules/Proj0800.md:10]: You are talking about DevelopmentDependency here, but in the implementation PR it is about ManagePackageVersionsCentrally?

File: rules/Proj0800.md (hunk 3 above)
Comment (Corniel) [reply rules/Proj0800.md:12]: Oops that is 26 times!

File: rules/Proj0800.md (hunk 5 above)
Comment (Corniel) [reply rules/Proj0800.md]: I think that what you refer to might be an interesting new issue and/or rule. But it should have been `<ManagePackageVersionsCentrally>`. Has been fixed.

Comment (Corniel) [reply rules/Proj0800.md]: Well spotted. Should have been `<ManagePackageVersionsCentrally>`.

**Review by CptWesley — APPROVED** [review]
LGTM

**Review by Sjaaky — CHANGES_REQUESTED** [review]
see comments

File: rules/Proj0800.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Proj0800: Configure Central Package Management explicitly
+Dependency management is a core feature of NuGet. Managing dependencies for a
+single project can be easy. Managing dependencies for multi-project solutions
+can prove to be difficult as they start to scale in size and complexity. In
+situations where you manage common dependencies for many different projects,
+you can leverage Central Package Management (CPM) features to do all of
+this from the ease of a single location.
+
+Hence, this rules advises to enable Central Package Management by
+enabling it with `<ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>`.
```
[the remark this answers is not in the document]
Comment (Corniel) [reply rules/Proj0800.md:10]: It does not state that it should be in the property file. (There will be another rule that ensures that it is enabled in the Packages.props, and not somewhere else).

I added a line to make clear that is advised to enable it the `Directory.Packages.props` file.

File: rules/Proj0800.md [hunk 8]
```diff
@@ -7,41 +7,26 @@ you can leverage Central Package Management (CPM) features to do all of
...
- <PropertyGroup>
-   <TargetFramework>net8.0</TargetFramework>
-  </PropertyGroup>
- 
-</Project>
-```
-
-## Compliant
 ``` XML
 <Project Sdk="Microsoft.NET.Sdk">
```
Comment (Sjaaky) [inline rules/Proj0800.md:13]: No `Sdk` in `Directory.Packages.props`

File: rules/Proj0800.md [hunk 9]
```diff
@@ -7,41 +7,26 @@ you can leverage Central Package Management (CPM) features to do all of
...
  </PropertyGroup>
  
 </Project>
 ```
 
 When a solution has only a tiny number of project files, and/or hardly any
-package is shared amongst these projects, CPM might be not worth the
-effort. In that case it is also fine to explicitly disable it.
+packages shared amongst these projects, CPM might be not worth the
+effort. In that case it is also fine to explicitly disable it:
```
Comment (Sjaaky) [inline rules/Proj0800.md:24]: (.. disable it ) in the project file:

File: rules/Proj0800.md [hunk 10]
```diff
@@ -7,41 +7,26 @@ you can leverage Central Package Management (CPM) features to do all of
...
 
 ``` XML
 <Project Sdk="Microsoft.NET.Sdk">
 
  <PropertyGroup>
-   <TargetFramework>net8.0</TargetFramework>
```
Comment (Sjaaky) [inline rules/Proj0800.md]: TargetFramework is valid in a project file. We might want to keep it here, to make it more clear that it is about the project file.

**Review by Sjaaky — APPROVED** [review]
👍

**Pull request merged**